python/private/platform_info.bzl

```diff
@@ -0,0 +1,34 @@
+"""Helper to define a struct used to define platform metadata."""
+
+def platform_info(
+        *,
+        compatible_with = [],
+        flag_values = {},
+        target_settings = [],
+        os_name,
+        arch):
+    """Creates a struct of platform metadata.
+
+    This is just a helper to ensure structs are created the same and
+    the meaning/values are documented.
+
+    Args:
+        compatible_with: list[str], where the values are string labels. These
+            are the target_compatible_with values to use with the toolchain
+        flag_values: dict[str|Label, Any] of config_setting.flag_values
+            compatible values. DEPRECATED -- use target_settings instead
+        target_settings: list[str], where the values are string labels. These
+            are the target_settings values to use with the toolchain.
+        os_name: str, the os name; must match the name used in `@platfroms//os`
```

Comment:
The arch and os could be retrieved from constraints list on one hand, on the other hand we could cconstruct the constraints from the os and arch.

Reply:
Looking at CI, we might have to change the tag class to accept an os_name and arch value, which I don't really like. I like the simplicity of a list of opaque labels. It is a simple list, though, so we could just look for the common @platforms values and infer them.

The problem CI is showing is, when `{name}_host` goes to be generated, it looks at the global PLATFORMS variable. That's easy to fix, but then it wants to look at the os_name and arch fields of that platform metadata. For these "ad hoc" platforms, those values weren't set, so then it blows up.

I'm going to try and see if I can make it skip generating a host repo for now. I'm still tring to figure out this host-repo code, because it seems like it creates a host repo for every registered version (full version).

Reply:
Yeah, this is for the pip integration only. (Some users may have some custom repo rules that also rely on this).